WordPress/gutenberg
# Post Featured Image block: Enable gradient overlay

## What?

- Adds the `Overlay` settings (color & opacity) to the Post featured block.

## Why?
- Checkout: https://github.com/WordPress/gutenberg/issues/43676 & https://github.com/WordPress/twentytwentythree/issues/30


## How?


## Testing scenarios covered
- Default gradients (Both inside site editor & the post editor)
- Custom gradients (Both inside site editor & the post editor)
- Default solid overlay color (Both inside site editor & the post editor)
- Custom solid overlay color (Both inside site editor & the post editor)
- Used inside the Query loop
- Didn't have a featured image on a post. (Placeholder will show the overlay as well)
- Above all at custom opacity levels and at 0.
- Above customizations with the Duotone applied. (Behaves same as the Cover block, i.e. applies an overlay on top of duotoned image).
- Overlay with the border settings applied.

## Testing Instructions




1. Open the post editor or the site editor.
1. Insert the `Post Featured Block`.
1. You should see the `Overlay` controls in the inspector controls of the block.
1. Applying the overlay should reflect both in the editor and on the front end.

## Screenshots or screencast 
![Sep-03-2022 00-15-46](https://user-images.githubusercontent.com/59014930/188218200-38c5465f-c0a8-40eb-b9ac-2191a2bc9e6d.gif)

## Important notes
- When combined with the Duotone, it behaves similarly to the Cover block.
- The Cover block doesn't show the overlay on the placeholder inside the site editor. Either we add the overlay on the placeholder for the cover block, too, or we remove it from Post featured block to ensure consistent UX.
